metricflow/model/data_warehouse_model_validator.py

```diff
@@ -130,46 +161,60 @@ def gen_dimension_tasks(cls, model: UserConfiguredModel) -> List[DataWarehouseVa
         on the data source to identify which have issues.
         """
 
+        render_tools = QueryRenderingTools(model=model, system_schema=system_schema)
+
         tasks: List[DataWarehouseValidationTask] = []
-        for index, data_source in enumerate(model.data_sources):
-            # generate the subtasks
-            data_source_tasks: List[DataWarehouseValidationTask] = []
-            data_source_columns: List[str] = []
-            for dimension in data_source.dimensions:
-                dim_to_query = dimension.expr if dimension.expr is not None else dimension.name
-                data_source_columns.append(dim_to_query)
-
-                data_source_tasks.append(
+        for data_source in model.data_sources:
+            if len(data_source.dimensions) > 0:
+                source_node = cls._data_source_node(render_tools=render_tools, data_source=data_source)
+
+                data_source_sub_tasks: List[DataWarehouseValidationTask] = []
+                data_source_specs: List[DimensionSpec] = []
+                for dimension in data_source.dimensions:
+                    if dimension.type == DimensionType.TIME:
+                        spec = TimeDimensionSpec.from_name(name=dimension.name)
+                    else:
+                        spec = DimensionSpec.from_name(name=dimension.name)
+                    data_source_specs.append(spec)
+                    filter_elements_node = FilterElementsNode(parent_node=source_node, include_specs=[spec])
```

Comment:
Let's use the existing converters for initializing these dimension specs - the process might change in the future and cause validation to fail in some real world cases but not others, so this will protect us from that.
```suggestion
                data_source_sub_tasks: List[DataWarehouseValidationTask] = []
                dataset = render_tools.converter.create_sql_source_dataset(data_source)
                data_source_specs = dataset.instance_set.spec_set.dimension_specs + dataset.instance_set.spec_set.time_dimension_specs
                for spec in data_source_specs:
                    filter_elements_node = FilterElementsNode(parent_node=source_node, include_specs=[spec])
```